File: documentation/manual/working/scalaGuide/advanced/routing/code/scalaguide/binder/controllers/BinderApplication.scala
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package scalaguide.binder.controllers

import javax.inject.Inject

import play.api.mvc._
import scalaguide.binder.models._

class BinderApplication @Inject() (components: ControllerComponents) extends AbstractController(components) {
  // #path
  def user(user: User) = Action {
    Ok(user.name)
  }
  // #path

  // #query
  def age(age: AgeRange) = Action {
    Ok(age.from.toString)
  }
  // #query
}

File: documentation/manual/working/javaGuide/advanced/routing/code/javaguide/binder/controllers/BinderApplication.java
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package javaguide.binder.controllers;

import javaguide.binder.models.AgeRange;
import javaguide.binder.models.User;
import play.mvc.Controller;
import play.mvc.Result;

public class BinderApplication extends Controller {

  // #path
  public Result user(User user) {
    return ok(user.name);
  }
  // #path

  // #query
  public Result age(AgeRange ageRange) {
    return ok(String.valueOf(ageRange.from));
  }
  // #query
}
